### PR TITLE
Add Imviz documentation

### PR DESCRIPTION
**Pull request by javerbukh**
### Description

This pull request adds documentation for Imviz.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:14]: Just one comment for now: Please use Sphinx `:ref:` for internal linking.

I'll review more tomorrow.

Comment (javerbukh) [reply docs/imviz/displayimages.rst]: Done.

File: docs/imviz/index.rst
Comment (pllim) [inline docs/imviz/index.rst:24]: I cannot comment on the lines above but I think this paragraph on L17-L18 also needs some rephrasing... 😅 

`Imviz allows images to be easily displayed and examined. It supports WCS and so on. (TODO: Add content.)`

File: docs/imviz/import_data.rst
Comment (pllim) [inline docs/imviz/import_data.rst:7]:
```suggestion
Imviz can load data in the form of a filename (FITS, JPEG, or PNG),
```

File: docs/imviz/import_data.rst
Comment (pllim) [inline docs/imviz/import_data.rst:8]:
```suggestion
an `~astropy.nddata.NDData` object, or a NumPy array if the data is 2D.
See :meth:`jdaviz.configs.imviz.helper.Imviz.load_data` for more information.
```

File: docs/imviz/import_data.rst
Comment (pllim) [inline docs/imviz/import_data.rst:16]: It only works on 2D, yes, but it is not limited to NDData, so it is misleading to put the parenthesis here. I think this was already covered in the section above anyway.

```suggestion
that can be parsed as a :class:`~astropy.nddata.NDData`, :class:`~astropy.io.fits.HDUList`,
```

File: docs/imviz/import_data.rst
Comment (pllim) [inline docs/imviz/import_data.rst:19]: I think @orifox said no need for screenshot. It is misleading to show Cubeviz here. Imviz user will see a different background and then they will be confused. If you must, refer back to relevant Cubeviz section with `:ref:` (as you have done below) but remove the image embedding.

```suggestion
or :class:`~astropy.io.fits.ImageHDU` in the text field.
```

File: docs/imviz/import_data.rst
Comment (pllim) [inline docs/imviz/import_data.rst:24]:
```suggestion
tab of each viewer's options menu as described in :ref:`cubeviz-selecting-data`.
```

File: docs/imviz/import_data.rst
Comment (pllim) [inline docs/imviz/import_data.rst:31]:
```suggestion
users can load data into the application through code using the :meth:`~jdaviz.configs.imviz.helper.Imviz.load_data`
```

File: docs/imviz/import_data.rst
Comment (pllim) [inline docs/imviz/import_data.rst:32]:
```suggestion
method, which takes as input either the name of a local file, an
```

File: docs/imviz/import_data.rst
Comment (pllim) [inline docs/imviz/import_data.rst:33]:
```suggestion
:class:`~astropy.nddata.NDData`, :class:`~astropy.io.fits.HDUList`,
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:7]:
```suggestion
Imviz uses image viewers to visualize data from supported formats.
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:16]:
```suggestion
        Documentation on selecting data sets in the Jdaviz viewers.
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:24]:
```suggestion
that has been loaded into the application to be visualized in these additional viewers. You can then utilize some of the
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:34]:
```suggestion
    :ref:`Defining Spatial Regions <spatial-regions>`
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:43]: Uppercase might make user think they need to do Shift+b when this is not the case.
```suggestion
by selecting the "b" key on your keyboard.
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:50]: This will be true after we merge #735.
```suggestion
images in multiple different viewers simultaneously. This works by matching images
based on the way they are linked together. Images are linked by pixels on load time,
but you can re-link them via WCS using `~jdaviz.configs.imviz.helper.Imviz.link_data`.
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:52]:
```suggestion
Regular Pan/Zoom is also available and is used in a similar way as other Jdaviz tools.
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:57]:
```suggestion
        Documentation on using Pan/Zoom in the Jdaviz viewers.
```

**Review by pllim — COMMENTED** [review]
In `displayimages.rst`, do you also want to mention:

* the cursor info bar on the upper right that displays cursor locations (pixel, RA+Dec) and value?
* the contrast/bias button?

In `notebook.rst`, we can also mention the `astrowidgets` API. But if you don't want to deal with that, I can open a follow-up PR after this one is merged. Up to you.

**Comment by pllim** [comment]
Also, overall, good work. Thank you very much! 👏

File: docs/imviz/import_data.rst
Comment (rosteen) [inline docs/imviz/import_data.rst:13]: In `displaying_images` you addressed the reader as "you", but here you're talking about "the user" in the abstract. I think you should use the former here for consistency.

File: docs/imviz/index.rst
Comment (javerbukh) [reply docs/imviz/index.rst:24]: Haha, good catch!

**Comment by javerbukh** [comment]
@pllim I did not mention `astrowidgets` since I think that was out of scope for this PR. Otherwise, thank you (and @rosteen ) for your edits!

File: docs/imviz/displayimages.rst
Comment (rosteen) [inline docs/imviz/displayimages.rst:9]: The Cubeviz section this links to is out of date, but I guess updating that can be a separate PR.

File: docs/imviz/displayimages.rst
Comment (rosteen) [inline docs/imviz/displayimages.rst:21]: Any chance you can just include a small image of the icon here? "picture with +" is very unwieldy compared to just showing the icon.

File: docs/imviz/displayimages.rst
Comment (rosteen) [inline docs/imviz/displayimages.rst:41]:
```suggestion
icon and then clicking the :guilabel:`right arrow` icon. You can also blink
```

File: docs/imviz/displayimages.rst
Comment (rosteen) [inline docs/imviz/displayimages.rst:40]: Is it two only, or two or more? Might be worth saying whether the images to blink between need to be selected in the data menu or not.

File: docs/imviz/displayimages.rst
Comment (rosteen) [inline docs/imviz/displayimages.rst:48]: Same comment as above, including the icon itself here might be nice if a small image can be included inline.

File: docs/imviz/displayimages.rst
Comment (rosteen) [inline docs/imviz/displayimages.rst:51]: Which direction changes which property?

File: docs/imviz/displayimages.rst
Comment (rosteen) [inline docs/imviz/displayimages.rst:59]:
```suggestion
of the data there. This information is located in the top bar of the UI, on the
```

**Review by rosteen — COMMENTED** [review]
Added a few more comments and suggestions. None of them are dealbreakers, I'll approve after you accept/decide to ignore them.

File: docs/imviz/displayimages.rst
Comment (javerbukh) [reply docs/imviz/displayimages.rst:51]: I think that's a question for @pllim , I'd have to spend some time figuring it out.

Comment (pllim) [reply docs/imviz/displayimages.rst:51]: https://github.com/spacetelescope/jdaviz/blob/79e4245663691a5f6567993be4c0198edebf4fbb/jdaviz/configs/imviz/plugins/tools.py#L150-L152

Drag across X changes bias, Y contrast. 😸

**Comment by javerbukh** [comment]
@rosteen I have to do some chores before I leave for class today but I should have these edits in by tonight/early tomorrow.

File: docs/imviz/displayimages.rst
Comment (javerbukh) [reply docs/imviz/displayimages.rst:40]: @pllim ? I was not able to blink between more than two images, not sure if that is user error or not.

File: docs/imviz/displayimages.rst
Comment (javerbukh) [reply docs/imviz/displayimages.rst:48]: Do you know how to add small images inline? I am only familiar with adding images to .rst files with `.. image::`.

**Comment by javerbukh** [comment]
@rosteen I included images for the `add viewer` and `contrast/bias` sections, although they are not inline. I'm waiting to hear from @pllim if blinking can be done with more than 2 images but my attempts were unsuccessful. I agree that updating the cubeviz section is part of a new ticket/PR.

**Comment by pllim** [comment]
> if blinking can be done with more than 2 images but my attempts were unsuccessful

Yes, it should be able to blink all the images as long as they are all linked. How was it unsuccessful for you?

**Comment by javerbukh** [comment]
I tried to load a PNG that was unrelated to the first two files, which was probably the issue. I will update the documentation then, thank you!

**Comment by pllim** [comment]
> I tried to load a PNG that was unrelated to the first two files

Yeah, you would need to relink everything for blink to work, but hopefully all this is automatic once #735 is merged. I think your doc update looks good either way. I'll re-review tomorrow. Thanks!

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:25]: I prefer the image to not break up the paragraph.

```suggestion
.. image:: img/picture_with_plus.png

In the toolbar towards the top of the UI, there is a :guilabel:`picture with a plus` icon
that when clicked will add new viewers to the application. You can then select from the data
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:71]:
```suggestion
.. _imviz_pan_zoom:

Pan/Zoom and Linked Pan/Zoom
============================
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:27]:
```suggestion
You can then utilize some of the Imviz-specific features, like :ref:`imviz_pan_zoom`.
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:43]: I think this is good enough, unless you want a new section on Imviz linking?
```suggestion
between viewing two or more images, as long as they are linked
(see :ref:`imviz_pan_zoom` for more on linking behavior). This can be done
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:59]: Can never be clear enough... 😆 

```suggestion
double-click on the display while the mode is active.
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:55]:
```suggestion
.. image:: img/white_to_black.png

In addition to changing contrast and bias information in the :guilabel:`layer` tab
(see :ref:`Contrast <contrast>`), Imviz has a :guilabel:`white to black` button
under the :guilabel:`hammer and screwdriver` menu that can also edit those options. After clicking
```

File: docs/imviz/displayimages.rst
Comment (pllim) [inline docs/imviz/displayimages.rst:57]: You capitalized X and Y somewhere else, so might as well be consistent.
```suggestion
and bias. Moving along the X-axis will change the bias and moving along the Y-axis will change the
```

**Review by pllim — COMMENTED** [review]
Some final comments, otherwise this is much better than I would have written. Great job! 👏

**Comment by javerbukh** [comment]
@pllim That's not true but I appreciate you saying it 😄

**Review by pllim — APPROVED** [review]
LGTM. Thanks! I'll let @rosteen have another look too, if he wants.

**Review by rosteen — APPROVED** [review]

**Pull request merged**
